File: httpsnippet/src/test/resources/output/node/unirest/multipart-file.js
var fs = require("fs");
var unirest = require("unirest");

var req = unirest("POST", "http://mockbin.com/har");

req.headers({"content-type":"multipart/form-data"});

req.multipart([{"body ":"fs.createReadStream(\"hello.txt\")","content-type":"text/plain"}]);

req.end(function (res) {
  if (res.error) throw new Error(res.error);

  console.log(res.body);
});

File: httpsnippet/src/test/resources/output/objc/nsurlsession/multipart-file.m
#import <Foundation/Foundation.h>

NSDictionary *headers = @{ @"content-type": @"multipart/form-data" };

NSData *postData = [[NSData alloc] initWithData:[@"" dataUsingEncoding:NSUTF8StringEncoding]];

NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:[NSURL URLWithString:@"http://mockbin.com/har"]
                                                       cachePolicy:NSURLRequestUseProtocolCachePolicy
                                                   timeoutInterval:10];
[request setHTTPMethod:@"POST"];
[request setAllHTTPHeaderFields:headers];
[request setHTTPBody:postData];

NSURLSession *session = [NSURLSession sharedSession];
NSURLSessionDataTask *dataTask = [session dataTaskWithRequest:request
                                            completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
                                              if (error) {
                                                NSLog(@"%@", error);
                                              } else {
                                                NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *) response;
                                                NSLog(@"%@", httpResponse);
                                              }
                                            }];
[dataTask resume];